<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
  <channel>
    <title>UpdateManager on Clarity</title>
    <link>https://jgautierit.github.io/tags/updatemanager/</link>
    <description>Recent content in UpdateManager on Clarity</description>
    <generator>Hugo -- gohugo.io</generator>
    <copyright>Copyright © 2018–2021, Jonathan Gautier; all rights reserved.</copyright>
    <lastBuildDate>Tue, 23 Jun 2020 00:00:00 +0000</lastBuildDate><atom:link href="https://jgautierit.github.io/tags/updatemanager/index.xml" rel="self" type="application/rss+xml" />
    <item>
      <title>vSphere 7 Lifecycle Manager - Can&#39;t Sync the HCL</title>
      <link>https://jgautierit.github.io/post/lifecycle-manager-cant-sync-the-hcl/</link>
      <pubDate>Tue, 23 Jun 2020 00:00:00 +0000</pubDate>
      
      <guid>https://jgautierit.github.io/post/lifecycle-manager-cant-sync-the-hcl/</guid>
      <description>
        
          &lt;p&gt;After upgrading vCenter 6.7 to 7.0 for one of my customer, I had a strange issue where I could not Sync the HCL with the Lifecycle Manager in vCenter 7.0&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/synchcl.jpg&#34; alt=&#34;LCM sync HCL&#34;&gt;&lt;/p&gt;
&lt;p&gt;In my situation, the task was stuck at 10% and failing afterwards with the error &amp;quot;A general system error occcured...&amp;quot;&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/errorlcm.jpg&#34; alt=&#34;Error HCL&#34;&gt;&lt;/p&gt;
&lt;p&gt;Looking at the logs (/var/log/vmware/vmware-updatemgr/vum-server/vmware-vum-server.log), it was really difficult to find something relevant but it looks like the download was failing.&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-bash&#34; data-lang=&#34;bash&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;20-06-23T10:09:55.200Z error vmware-vum-server&lt;span class=&#34;o&#34;&gt;[&lt;/span&gt;28017&lt;span class=&#34;o&#34;&gt;]&lt;/span&gt; &lt;span class=&#34;o&#34;&gt;[&lt;/span&gt;Originator@6876 &lt;span class=&#34;nv&#34;&gt;sub&lt;/span&gt;&lt;span class=&#34;o&#34;&gt;=&lt;/span&gt;com.vmware.vcIntegrity.hcl.UpdateHclDatastoreTask&lt;span class=&#34;o&#34;&gt;]&lt;/span&gt; Task:com.vmware.vcIntegrity.hcl.UpdateHclDatastoreTask ID:52b92e6b-1b4c-1584-c1ad-da57f7571f88. Task Failed. Error: Error:
&lt;span class=&#34;ln&#34;&gt;2&lt;/span&gt;--&amp;gt;    com.vmware.vapi.std.errors.error
&lt;span class=&#34;ln&#34;&gt;3&lt;/span&gt;--&amp;gt; Messages:
&lt;span class=&#34;ln&#34;&gt;4&lt;/span&gt;--&amp;gt;    com.vmware.vcIntegrity.hcl.vvs.unreachable.summary
&lt;span class=&#34;ln&#34;&gt;5&lt;/span&gt;--&amp;gt;    com.vmware.vcIntegrity.hcl.vvs.unreachable.resolution
&lt;span class=&#34;ln&#34;&gt;6&lt;/span&gt;--&amp;gt;
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;Looking at the tasks, I found out that VMware use multile python scripts in the following directory   /usr/lib/vmware-updatemgr/config/&lt;/p&gt;
&lt;p&gt;Thanks to my buddy Y.Ilmi for guiding me to the specific script called vvs-config.json.&lt;/p&gt;
&lt;p&gt;This script provides the URL that VMware use to download the hcl : vvs.esp.vmware.com&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/vvsconfigjson.jpg&#34; alt=&#34;vvs-config&#34;&gt;&lt;/p&gt;
&lt;p&gt;vCenter was unable to reach the URL (use curl to test it) I asked the security team to grant access.&lt;/p&gt;
&lt;p&gt;Problem solved!&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/HCL2.jpg&#34; alt=&#34;HCL&#34;&gt;&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/hcl3.jpg&#34; alt=&#34;HCL&#34;&gt;&lt;/p&gt;

        
      </description>
    </item>
    
    <item>
      <title>vSphere 7 Lifecycle Manager - Can&#39;t Sync Updates</title>
      <link>https://jgautierit.github.io/post/lifecycle-manager-cant-sync-update/</link>
      <pubDate>Thu, 04 Jun 2020 00:00:00 +0000</pubDate>
      
      <guid>https://jgautierit.github.io/post/lifecycle-manager-cant-sync-update/</guid>
      <description>
        
          &lt;p&gt;After upgrading vCenter 6.7 to 7.0, I had a strange issue where I could not Sync Updates anymore within the Lifecycle Manager in vCenter 7.0&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/lcmsync.jpg&#34; alt=&#34;LCM sync&#34;&gt;&lt;/p&gt;
&lt;p&gt;The error message was the following : A general system occured: An error occured while downloading depot metadata from file:///storage/updatemgr/patch-store/hostupdate/vmw/vmw-ESXi-5.0.0-metadata.zip&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;https://jgautierit.github.io/lcmerror.jpg&#34; alt=&#34;LCM error&#34;&gt;&lt;/p&gt;
&lt;p&gt;From the error, it looks like it&#39;s trying to download the metadata file from the 5.0 zip file which should not be really used anymore since vCenter 7.0 is not compatible anymore with any hosts &amp;lt; 6.5.&lt;/p&gt;
&lt;p&gt;I went to look at the directory (/storage/updatemgr/patch-store/hostupdate/vmw/) within vCenter and indeed the file was missing. In order to see if I was really missing a file here, I went to look in a freshly deployed vCenter and it seems like there is no file named this way either.&lt;/p&gt;
&lt;p&gt;I tried to compare the xml files and see if anything was missing or corrupted but everything seemed to be fine.&lt;/p&gt;
&lt;p&gt;At some point, I decided to reset completely Update Manager but I was not sure how.&lt;/p&gt;
&lt;p&gt;I found this VMware Knowledge Base article &lt;a href=&#34;https://kb.vmware.com/s/article/2147284&#34; title=&#34;VMware KB2147284&#34;&gt;here&lt;/a&gt; which describe how to reset the database.&lt;/p&gt;
&lt;p&gt;After executing the following steps my issue was fixed and update manager was not complaining about the file anymore (even though it&#39;s still does not exist...)&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-bash&#34; data-lang=&#34;bash&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;&lt;span class=&#34;c1&#34;&gt;# service-control --start vmware-updatemgr&lt;/span&gt;
&lt;span class=&#34;ln&#34;&gt;2&lt;/span&gt;&lt;span class=&#34;c1&#34;&gt;# /usr/lib/vmware-updatemgr/bin/updatemgr-utility.py reset-db&lt;/span&gt;
&lt;span class=&#34;ln&#34;&gt;3&lt;/span&gt;&lt;span class=&#34;c1&#34;&gt;# rm -rf /storage/updatemgr/patch-store/*&lt;/span&gt;
&lt;span class=&#34;ln&#34;&gt;4&lt;/span&gt;&lt;span class=&#34;c1&#34;&gt;# service-control --start vmware-updatemgr&lt;/span&gt;
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;I hope this has been useful for you and I&#39;d like to thank you for reading!&lt;/p&gt;

        
      </description>
    </item>
    
    <item>
      <title>Update Manager 6.7 fails to remediate - Check your ports !</title>
      <link>https://jgautierit.github.io/post/update-manager-fails-to.remediate/</link>
      <pubDate>Thu, 20 Sep 2018 00:00:00 +0000</pubDate>
      
      <guid>https://jgautierit.github.io/post/update-manager-fails-to.remediate/</guid>
      <description>
        
          &lt;p&gt;This quick thread will relate issues regarding the remediation of my upgrade to 6.7.
Working at a customer site and upgrading the hosts to 6.7, Update Manager was failing to remediate my ESXi upgrade with the following error :&lt;/p&gt;
&lt;p&gt;&lt;img src=&#34;vumerror.png&#34; alt=&#34;Remediate Error&#34;&gt;&lt;/p&gt;
&lt;p&gt;What a nice error message ...Very Helpful!&lt;/p&gt;
&lt;p&gt;No more indication was provided by the vCenter tasks &amp;amp; events so I had to look at the logs.&lt;/p&gt;
&lt;p&gt;In vCenter Server Appliance, the Update manager logs are located here :&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-bash&#34; data-lang=&#34;bash&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;/var/log/vmware/vmware-updatemgr/vum-server
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;Looking at the last &amp;quot;vmware-vum-server.log&amp;quot;, I quickly matched an error with the last try :&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-fallback&#34; data-lang=&#34;fallback&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;2018-09-20T11:31:14.436Z warning vmware-vum-server[06197] [Originator@6876 sub=Default] Failed to connect socket; &amp;lt;io_obj p:0x00007fb0840161f0, h:20, &amp;lt;TCP &amp;#39;X.X.X.X : 42026&amp;#39;&amp;gt;, &amp;lt;TCP &amp;#39;Y.Y.Y.Y : 80&amp;#39;&amp;gt;&amp;gt;, e: 111(Connection refused)
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;This error indicates that the IP X.X.X.X tries to communicate through the port 80 on the IP address Y.Y.Y.Y. X being my vCenter and Y my ESXi host.&lt;/p&gt;
&lt;p&gt;In order to prove that the error was the consequence of a blocked port, I used this the following command to test the ports :&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-bash&#34; data-lang=&#34;bash&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;&lt;span class=&#34;c1&#34;&gt;# curl -v telnet://Y.Y.Y.Y:80&lt;/span&gt;
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;Guess the result ?&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-fallback&#34; data-lang=&#34;fallback&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;* Rebuilt URL to: telnet://Y.Y.Y.Y:80/
&lt;span class=&#34;ln&#34;&gt;2&lt;/span&gt;*   Trying Y.Y.Y.Y...
&lt;span class=&#34;ln&#34;&gt;3&lt;/span&gt;* TCP_NODELAY set
&lt;span class=&#34;ln&#34;&gt;4&lt;/span&gt;* connect to Y.Y.Y.Y port 80 failed: Connection refused
&lt;span class=&#34;ln&#34;&gt;5&lt;/span&gt;* Failed to connect to Y.Y.Y.Y port 80: Connection refused
&lt;span class=&#34;ln&#34;&gt;6&lt;/span&gt;* Closing connection 0
&lt;span class=&#34;ln&#34;&gt;7&lt;/span&gt;curl: (7) Failed to connect to Y.Y.Y.Y port 80: Connection refused
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;We then tested a known working port such as 443 :&lt;/p&gt;
&lt;div class=&#34;highlight&#34;&gt;&lt;pre class=&#34;chroma&#34;&gt;&lt;code class=&#34;language-fallback&#34; data-lang=&#34;fallback&#34;&gt;&lt;span class=&#34;ln&#34;&gt;1&lt;/span&gt;# curl -v telnet://Y.Y.Y.Y:443
&lt;span class=&#34;ln&#34;&gt;2&lt;/span&gt;* Rebuilt URL to: telnet://Y.Y.Y.Y:443
&lt;span class=&#34;ln&#34;&gt;3&lt;/span&gt;*   Trying Y.Y.Y.Y...
&lt;span class=&#34;ln&#34;&gt;4&lt;/span&gt;* TCP_NODELAY set
&lt;span class=&#34;ln&#34;&gt;5&lt;/span&gt;* Connected to Y.Y.Y.Y (Y.Y.Y.Y) port 443 (#0)
&lt;/code&gt;&lt;/pre&gt;&lt;/div&gt;&lt;p&gt;It become then obvious that the port was blocked by some firewall.&lt;/p&gt;
&lt;p&gt;We asked kindly the colleagues to open the port. Required ports for update manager are listed &lt;a href=&#34;https://kb.vmware.com/s/article/1004543&#34; title=&#34;KB1004543&#34;&gt;here&lt;/a&gt;:&lt;/p&gt;
&lt;p&gt;Problem Solved!&lt;/p&gt;

        
      </description>
    </item>
    
  </channel>
</rss>
